src/main/java/practice/CandidateValidator.java

```
@@ -1,5 +1,31 @@
 package practice;
 
-public class CandidateValidator {
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate.getAge() < 35) {
+            return false;
+        }
+        if (!candidate.isAllowedToVote()) {
+            return false;
+        }
+        if (!candidate.getNationality().equals("Ukrainian")) {
+            return false;
+        }
+        return getPeriodInUkraineInt(candidate.getPeriodsInUkr()) >= 10;
+    }
+
+    private int getPeriodInUkraineInt(String period) {
+        if (period != null && !period.isEmpty()) {
+            String[] years = period.split("-");
+            return years.length > 1
+                    ? Integer.parseInt(years[1]) - Integer.parseInt(years[0]) + 1
+                    : Integer.parseInt(years[0]);
```

Don’t provide your own logic. If the parameter period doesn’t  have format `number-number` you should throw an exception